xrrandom/sampling.py

```diff
@@ -80,7 +80,7 @@ def generate_virtual_samples(gen_func, args, samples:int=1, output_dtype=np.floa
         perform any operations and call compute() after changing the number of samples.
     """
     if sample_chunksize is None:
-        samples_arr = da.from_array(np.full((samples,), samples), name=SAMPLE_VEC_KEY)
+        samples_arr = da.from_array(np.full((samples,), samples), name=SAMPLE_VEC_KEY, chunks='auto')
```

Comment:
`chunks='auto'` might create several chunks for large sample sizes which could potentially result in shape incompatibilities when computed, because for each chunk the full sample size would be generated. Perhaps a more sensible setting would be `chunks=samples`.